lib/flow/VaporField.cpp

```diff
@@ -48,26 +46,22 @@ VaporField::InsideVolumeVelocity( float time, const glm::vec3& pos ) const
         // Then test if pos is inside of time step "floor"
         for( auto& v : VelocityNames )
         {
-            if( !v.empty() )
-            {
-                grid = _getAGrid( floor, v );
-                VAssert( grid );
-                if( !grid->InsideGrid( coords ) )
-                    return false;
-            }
+            grid = _getAGrid( floor, v );
+            VAssert( grid );
```

Comment:
Should you be using VAssert(grid) here? It looks like _getAGrid can return NULL under conditions that are NOT programming errors. E.g. the variable is not available at the requested timestep

Reply:
Did you mean this place should not use an assert?

Reply:
asserts should only be used for programing errors. It would appear that you are using an assert here to address something that is not a programming error: a data read failure, which should have proper error handing.

Reply:
Fixed in commit c1ec340